cpp/src/arrow/array/diff.cc

```diff
@@ -822,6 +822,15 @@ class UnifiedDiffFormatter {
 
 Result<std::function<Status(const Array& edits, const Array& base, const Array& target)>>
 MakeUnifiedDiffFormatter(const DataType& type, std::ostream* os) {
+  if (type.id() == Type::NA) {
+    return [os](const Array& edits, const Array& base, const Array& target) {
+      *os << "# Null arrays differed" << std::endl
```

Comment:
Shouldn't you avoid outputting this if the too arrays are equal?

Reply:
Derp, yes